test/unit/alignment/matrix/trace_directions_test.cpp

```diff
@@ -0,0 +1,40 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <sstream>
+
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/alignment/matrix/trace_directions.hpp>
+
+using namespace seqan3;
+using namespace seqan3::detail;
+
+static constexpr trace_directions N = trace_directions::none;
+static constexpr trace_directions D = trace_directions::diagonal;
+static constexpr trace_directions U = trace_directions::up;
+static constexpr trace_directions L = trace_directions::left;
+
+TEST(debug_stream, ascii)
+{
+    std::stringstream s{};
+    debug_stream_type stream{s};
+    stream << N << ";" << D << ";" << U << ";" << L << ";" << (D|U) << ";" << (D|L) << ";" << (U|L) << ";" << (D|U|L);
+
+    EXPECT_EQ(s.str(), "N;D;U;L;DU;DL;UL;DUL");
```

Comment:
don't you need to flush the stream before using str()? I know we do that in other places to ensure that it works correctly I think

Reply:
Why should I need to flush? I tried to read up on this, I found a couple of "unofficial"  statements http://www.cplusplus.com/forum/general/84962/, http://forums.codeguru.com/showthread.php?292719-How-to-clear-out-a-stringstream&p=938707#post938707 and so on.

I mean it makes sense, you don't have to commit you intermediate data to some device. Which flushing is intended for.